### PR TITLE
add gas price from api

### PR DESCRIPTION
**Pull request by tien7668**

### CODE REVIEW

File: src/hooks/Trades.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { Currency, CurrencyAmount, Pair, Token, Trade } from '@dynamic-amm/sdk'
-import { useMemo, useEffect, useState, useCallback } from 'react'
+import { useMemo, useEffect, useState, useCallback, useReducer } from 'react'
```
Comment (viet-nv) [inline src/hooks/Trades.ts:2]: It's seem we do not use `useReducer` any where?

File: src/pages/App.tsx [hunk 2]
```diff
@@ -84,11 +88,40 @@ const Marginer = styled.div`
...
 export default function App() {
-  useActiveNetwork()
-  const { account } = useActiveWeb3React()
+  const { account, chainId } = useActiveWeb3React()
   const aboutPage = useRouteMatch('/about')
   const apolloClient = useExchangeClient()
-
+  const dispatch = useDispatch<AppDispatch>()
+  const fetchGas = (chain: string) => {
+    fetch(`https://staging-krystal-api.knstats.com/${chain}/v2/swap/gasPrice`)
```
Comment (viet-nv) [inline src/pages/App.tsx:96]: should we move this to env like: `REACT_APP_KRYSTAL_API = https://staging-krystal-api.knstats.com`

File: src/utils/aggregator.ts [hunk 3]
```diff
@@ -247,7 +247,8 @@ export class Aggregator {
     currencyAmountIn: CurrencyAmount,
     currencyOut: Currency,
     saveGas = false,
-    dexes = ''
+    dexes = '',
+    gasPrice: any
```
Comment (viet-nv) [inline src/utils/aggregator.ts:251]: Can we define gasPrice type here?

File: src/state/application/reducer.ts [hunk 4]
```diff
@@ -31,6 +32,7 @@ export interface ApplicationState {
   readonly kncPrice?: string
   readonly chainIdWhenNotConnected: ChainId
   exchangeSubgraphClients: { [key: string]: ApolloClient<NormalizedCacheObject> }
+  readonly gasPrice?: { [key: string]: string }
```
Comment (viet-nv) [inline src/state/application/reducer.ts:35]: we can define more clearly type for gas price here. I think it should be: { standard: string, fast: string, low: string, default: string }

**Review by viet-nv — CHANGES_REQUESTED** [review]

File: src/pages/App.tsx [hunk 5]
```diff
@@ -84,11 +88,40 @@ const Marginer = styled.div`
...
+  useEffect(() => {
+    let interval: any = null
+    const chain =
+      chainId == ChainId.MAINNET
+        ? 'ethereum'
+        : chainId == ChainId.BSCMAINNET
+        ? 'bsc'
+        : chainId == ChainId.AVAXMAINNET
+        ? 'avalanche'
+        : chainId == ChainId.MATIC
```
Comment (viet-nv) [inline src/pages/App.tsx:114]: Is Fantom supported?

File: src/hooks/Trades.ts (hunk 1 above)
Comment (tien7668) [reply src/hooks/Trades.ts]: agree, removed

File: src/pages/App.tsx (hunk 2 above)
Comment (tien7668) [reply src/pages/App.tsx]: agree

File: src/utils/aggregator.ts (hunk 3 above)
Comment (tien7668) [reply src/utils/aggregator.ts]: agree

File: src/state/application/reducer.ts (hunk 4 above)
Comment (tien7668) [reply src/state/application/reducer.ts]: agree

File: src/pages/App.tsx (hunk 5 above)
Comment (tien7668) [reply src/pages/App.tsx:114]: not yet

**Review by viet-nv — APPROVED** [review]

**Pull request merged**